### PR TITLE
Add new semantic markup support to ansible-doc text output and to validate-modules

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Implements ansible-doc part of #73137.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ansible-doc

### CODE REVIEW

File: lib/ansible/cli/doc.py [hunk 1]
```diff
@@ -348,10 +348,14 @@ class DocCLI(CLI, RoleMixin):
...
     _BOLD = re.compile(r"\bB\(([^)]+)\)")
     _MODULE = re.compile(r"\bM\(([^)]+)\)")
+    _PLUGIN = re.compile(r"\bP\(([^#)]+)#([a-z]+)\)")
     _LINK = re.compile(r"\bL\(([^)]+), *([^)]+)\)")
     _URL = re.compile(r"\bU\(([^)]+)\)")
     _REF = re.compile(r"\bR\(([^)]+), *([^)]+)\)")
     _CONST = re.compile(r"\bC\(([^)]+)\)")
+    _SEM_OPTION_NAME = re.compile(r"\bO\(([^)]+)\)")
+    _SEM_OPTION_VALUE = re.compile(r"\bV\(([^)]+)\)")
+    _SEM_ENV_VARIABLE = re.compile(r"\bE\(([^)]+)\)")
```
Comment (bcoca) [inline lib/ansible/cli/doc.py]: i would make these 3 into one (since they are basically same behavior)

Comment (felixfontein) [reply lib/ansible/cli/doc.py]: Combining `C()`, `V()` and `E()` makes senes, while I would keep `O()` differently. In the form `O()` without `=` in the parameter, the formatting should be bold (according to the IBM style guide), so if this is combined with your colors PR, this should be handled differently.

**Comment by felixfontein** [comment]
I rebased to latest `devel`. The community approved the plan of trying to get semantic markup into ansible-core (see https://github.com/ansible-community/community-topics/issues/53#issuecomment-1462559857), so I'm marking this as:

ready_for_review

:tada:

**Comment by mattclay** [comment]
@felixfontein Can you add some integration tests to cover this new functionality? The `ansible-test-sanity-validate-modules` and `ansible-doc` targets are probably the best places for the tests.

**Comment by samccann** [comment]
@mattclay @bcoca - since this is part of semantic markup support, do you feel this will be merged before 2.15 code freeze?

**Comment by felixfontein** [comment]
I can also split this up into two PRs - one for ansible-doc and one for validate-modules - if that helps.

**Comment by bcoca** [comment]
@samccann it is not waiting on us
@felixfontein probably, I can see us merging the ansible-doc changes w/o having impact on others, validate-modules part on the other hand ...

**Comment by felixfontein** [comment]
#80244 contains only the seealso schema change for validate-modules. That is the bare minimum of semantic markup support for validate-modules that is needed (the markup itself is not validated) so that one doesn't need to add ignore.txt entries when using this part of semantic markup.

**Comment by felixfontein** [comment]
Closing in favor of #80242 and #80243 or at least #80244.

**Pull request closed without merging**